src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+    
+    public final String value;
+    
+    private boolean isPrivate;
+    
+    public Contact(String contactDetail, boolean isPrivate) {
+        this.isPrivate = isPrivate;
+        this.value = contactDetail.trim();
+    }
+    
+    @Override
+    public String toString() {
+        return value;
+    }
+    
+    @Override
+    public boolean equals(Object other) {
```

Comment:
do take note of `equals` here.  If you check against `Contact` class type, then 

    Contact address =  new Address('something', false);
    Contact email = new Email('something', false);
    assert(address.equals(email)); // returns true.